server/metrics/statsd.go

```diff
@@ -0,0 +1,63 @@
+package metrics
+
+import (
+	"io"
+	"strings"
+	"time"
+
+	"github.com/cactus/go-statsd-client/statsd"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events/yaml/valid"
+	"github.com/runatlantis/atlantis/server/logging"
+	"github.com/uber-go/tally"
+	tallystatsd "github.com/uber-go/tally/statsd"
+)
+
+func NewLoggingScope(logger logging.SimpleLogging, statsNamespace string) (tally.Scope, io.Closer, error) {
+	reporter, err := newReporter(valid.Metrics{}, logger)
+
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "initializing stats reporter")
+	}
+
+	scope, closer := tally.NewRootScope(tally.ScopeOptions{
+		Prefix:   statsNamespace,
+		Reporter: reporter,
+	}, time.Second)
+
+	return scope, closer, nil
+}
+
+func NewScope(cfg valid.Metrics, logger logging.SimpleLogging, statsNamespace string) (tally.Scope, io.Closer, error) {
+	reporter, err := newReporter(cfg, logger)
+
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "initializing stats reporter")
+	}
+
+	scope, closer := tally.NewRootScope(tally.ScopeOptions{
+		Prefix:   statsNamespace,
+		Reporter: reporter,
+	}, time.Second)
+
+	return scope, closer, nil
+}
+
+func newReporter(cfg valid.Metrics, logger logging.SimpleLogging) (tally.StatsReporter, error) {
+	if cfg.Statsd == nil {
```

Comment:
Is this always the case? What if there was a misconfigation that somehow set the statsd to nil? Shouldn't we return an error here? Looks like `newLoggingReporter` is only used by tests.

Reply:
the idea is here is to have a set of fields in that config object for statsd, prometheus and really any time of metrics supported by tally.

We are checking to see which metrics object is populated and if nothing is, we just return the logging scope.